notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "sales_and_cluster_data = train_data_copy.drop(['date', 'family', 'onpromotion', 'city', 'type'], axis=1)\n",
+    "sales_and_cluster_data.corr()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figsize=(10, 7)\n",
+    "plt.scatter(x='cluster', y='sales', data=sales_and_cluster_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>'cluster' feature doesn't correlate with the target, and I can't see any dependecies between these 2 features. But I think that this feature can be useful, because it connects similar stores together.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 206
+    },
+    "id": "Tiavh7PK-HYv",
+    "outputId": "4d198de7-7161-4413-eefa-45b6201bd547"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data = pd.read_csv('C:\\\\Users\\\\Owner\\\\Desktop\\\\Proga\\\\REPOS\\\\mentorship_EPAM\\\\data\\\\oil.csv')\n",
+    "oil_data[\"time\"] = np.arange(len(oil_data.index))\n",
+    "oil_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "FDbzS7RpNnKK",
+    "outputId": "ef457345-a324-40da-db3e-6a7362248416"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 297
+    },
+    "id": "Bt59gF6ePlOZ",
+    "outputId": "9af1e0e1-f125-40ef-ba81-c9c889b7ba91"
+   },
+   "outputs": [],
+   "source": [
+    "oil_data_copy = oil_data.copy()\n",
+    "oil_data_copy.plot(kind=\"scatter\", x=\"time\", y=\"dcoilwtico\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = train_data.merge(oil_data, on='date', how=\"left\")\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.columns"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.drop(columns=['time'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = train_data.drop(['time'], axis=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_oil = train_data.fillna(train_data['dcoilwtico'].mean())\n",
```

Comment:
Why did you decise to fill missing values with the mean value in this case? Do you know any other ways of handling missing values in the dataset?

Reply:
The method I chose won't work well here, because for 4 years of observations the oil price has changed a lot, that is why it is a mistake to replace every NaN value with the mean value in 'dcoilwtico' column. There are a lot of ways of handling missing values in the particular column (they can be replaced by 0, **min**, or **max** value it the column, also custom values can be used). I changed my opinion and replaced all the NaN values in this column with values from shifted features (for example, I replaced Saturday's NaN value with Friday's value, and Sunday's NaN value with Monday's value). If there were 3 or 4 NaN values in a row, I used feature values ​​shifted forward and backward by 2 days). I think, that this technique is the most in this situation.